File: play_random_game.py
import random
from time import time

from tqdm import tqdm

from chessxtra_game import new_game, get_all_legal_moves, move_piece, colours

random.seed(1)

colours = colours()

def run_random_games(n: int = 1) -> None:
    outcomes = {colours[0]: 0, colours[1]: 0}
    for _ in tqdm(range(n)):
        game = new_game()
        # print(game.board)
        while not game.completed:
            legal_moves = get_all_legal_moves(
                game=game, colour=game.turn
            )

            chosen_piece = random.choice(list(legal_moves.keys()))
            chosen_move = random.choice(list(legal_moves[chosen_piece]))

            game = move_piece(game=game, pos_1=chosen_piece.position, pos_2=chosen_move)

            # print(game.board)
            # print()
        outcomes[game.winner] += 1
        p = 0
    print(outcomes)


def main() -> None:
    s = time()
    run_random_games(10_000)
    print(time() - s)


if __name__ == "__main__":
    main()

File: src/play_random_game.ts
const cliProgress = require('cli-progress');

function run_random_games(n: number): void {
    let outcomes = new Map()
    outcomes.set(Colour.White, 0)
    outcomes.set(Colour.Black, 0)
    const pbar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
    pbar.start(n, 0)
    for (const _ of range(n)) {
        let game = initialize_game()
        while (!game.completed) {
            const legal_moves = get_all_legal_moves(game, game.turn)
            const legal_move_pieces = [...legal_moves.keys()]
            const chosen_piece = legal_move_pieces[Math.floor(Math.random() * legal_move_pieces.length)]
            const chosen_move = legal_moves.get(chosen_piece)[Math.floor(Math.random() * legal_moves.get(chosen_piece).length)]
        
            move(game, chosen_piece.position, chosen_move)
        }
        outcomes.set(game.winner, outcomes.get(game.winner) + 1)
    }
    console.log(outcomes)
}

function main(): void {
    const startTime = performance.now()
    run_random_games(500)
    console.log(performance.now() - startTime)
}

main();
